### ISSUE TITLE
Not resolving module names

### ISSUE THREAD

**Issue by kxkayser** [namgk/node-red-contrib-pythonshell#11]
Hi. I am trying to use the Pythonshell node to start a program on a RPi. My default environment is Python 3.5.3. The program file has a shebang statement reflecting this environment  (#!/usr/bin/env python3) and works fine from the command line,  a desktop icon and the NR exec node. When executing it from the Pythonshell command, it can't find the first import module. It is finding the program. I assume it's launching the program in an environment other than where I have the modules installed but this may not really be the problem. I am quite new to this environment.

Using the "exec" node in NR works fine.  I am using exactly the same string as the payload: 
/home/pi/pp/dashboard/./temp_hum_pres_2.py

The error that I get is:
exit code: 1, Traceback (most recent call last):
  File "temp_hum_pres.py", line 4, in <module>
    import board
ImportError: No module named board

"board" is a module installed in my Python environment from Adafruit.

**Comment by namgk** [comment]
Yes sounds like an environment problem. The python script is spawned in a
child process environment that might not have some variables (e.g
PYTHONPATH) that lead it to the library to be imported.

Basically what the node does is just spawn a python process with the script
as the argument (i.e similar to doing $ python script.py)

In the meantime can you use the exec node?

Otherwise try to wrap your env into a virtual env, which is a common
practice in python. The node support virtual env as well.

 Cheers

**Comment by kxkayser** [comment]
Yes, I have used the exec node and it is working perfectly.  If the node spawns a child process that doesn’t have the PYTHONPATH variable, it’s probably not very useful in my opinion.  Just thought I should post the issue.

 

Thanks. 

 

Ken Kayser

Tobaccoville, NC

 

kxkayser@gmail.com

**Comment by namgk** [comment]
Thanks for bringing it up. I’ll look more into it.
